### PR TITLE
Normative: Change weekInfo to express non-continouse weekend

### PR DESCRIPTION
**Pull request by FrankYFTang**
Close #25

### CODE REVIEW

File: locale.html [hunk 1]
```diff
@@ -135,14 +135,9 @@ <h1>WeekInfoOfLocale ( _loc_ )</h1>
...
-            <td>The weekday value indicating which day of the week is considered the starting day of the 'weekend', for calendar purposes.</td>
-          </tr>
-          <tr>
-            <td>[[WeekendEnd]]</td>
-            <td>Integer value between 1 and 7: 1 specifies Monday; 2 specifies Tuesday; 3 specifies Wednesday; 4 specifies Thursday; 5 specifies Friday; 6 specifies Saturday; and 7 specifies Sunday.</td>
-            <td>The weekday value indicating which day of the week is considered the ending day of the 'weekend', for calendar purposes.</td>
+            <td>[[Weekend]]</td>
+            <td>A List of 1 or more Integer value between 1 and 7: 1 specifies Monday; 2 specifies Tuesday; 3 specifies Wednesday; 4 specifies Thursday; 5 specifies Friday; 6 specifies Saturday; and 7 specifies Sunday.</td>
+            <td>The list of weekday values indicating which days of the week are considered as part of the 'weekend', for calendar purposes. Notice the number of days in weekend are different in each locale and may not be continous.</td>
```
Comment (sffc) [inline locale.html:140]:
```suggestion
            <td>The list of weekday values indicating which days of the week are considered as part of the 'weekend', for calendar purposes. Notice that the number of days in the weekend are different in each locale, and they may not be contiguous.</td>
```

**Review by sffc — CHANGES_REQUESTED** [review]

File: annexes.html [hunk 2]
```diff
@@ -20,7 +20,7 @@ <h1>Implementation Dependent Behaviour</h1>
           The default general ordering of characters (characterOrder) within a line per locale (<emu-xref href="#sec-character-direction-of-locale"></emu-xref>)
         </li>
         <li>
-          The 'first' day of a week, the minimal days required in the first week of a month or year, and the weekday of the starting and ending day of the 'weekend', for calendar purposes, per locale (<emu-xref href="#sec-week-info-of-locale"></emu-xref>)
+          The 'first' day of a week, the minimal days required in the first week of a month or year, and days of the week are considered as part of the 'weekend', for calendar purposes, per locale (<emu-xref href="#sec-week-info-of-locale"></emu-xref>)
```
Comment (anba) [inline annexes.html:23]:
```suggestion
          The 'first' day of a week, the minimal days required in the first week of a month or year, and which days of the week are considered as part of the 'weekend', for calendar purposes, per locale (<emu-xref href="#sec-week-info-of-locale"></emu-xref>)
```

File: locale.html [hunk 3]
```diff
@@ -135,14 +135,9 @@ <h1>WeekInfoOfLocale ( _loc_ )</h1>
...
-            <td>The weekday value indicating which day of the week is considered the starting day of the 'weekend', for calendar purposes.</td>
-          </tr>
-          <tr>
-            <td>[[WeekendEnd]]</td>
-            <td>Integer value between 1 and 7: 1 specifies Monday; 2 specifies Tuesday; 3 specifies Wednesday; 4 specifies Thursday; 5 specifies Friday; 6 specifies Saturday; and 7 specifies Sunday.</td>
-            <td>The weekday value indicating which day of the week is considered the ending day of the 'weekend', for calendar purposes.</td>
+            <td>[[Weekend]]</td>
+            <td>A List of 1 or more Integer value between 1 and 7: 1 specifies Monday; 2 specifies Tuesday; 3 specifies Wednesday; 4 specifies Thursday; 5 specifies Friday; 6 specifies Saturday; and 7 specifies Sunday.</td>
+            <td>The list of weekday values indicating which days of the week are considered as part of the 'weekend', for calendar purposes. Notice that the number of days in the weekend are different in each locale, and they may not be contiguous.</td>
```
Comment (anba) [inline locale.html:140]:
```suggestion
            <td>The list of weekday values indicating which days of the week are considered as part of the 'weekend', for calendar purposes. Notice that the number of days in the weekend are different in each locale and may not be contiguous.</td>
```

File: locale.html [hunk 4]
```diff
@@ -135,14 +135,9 @@ <h1>WeekInfoOfLocale ( _loc_ )</h1>
...
-            <td>[[WeekendStart]]</td>
-            <td>Integer value between 1 and 7: 1 specifies Monday; 2 specifies Tuesday; 3 specifies Wednesday; 4 specifies Thursday; 5 specifies Friday; 6 specifies Saturday; and 7 specifies Sunday.</td>
-            <td>The weekday value indicating which day of the week is considered the starting day of the 'weekend', for calendar purposes.</td>
-          </tr>
-          <tr>
-            <td>[[WeekendEnd]]</td>
-            <td>Integer value between 1 and 7: 1 specifies Monday; 2 specifies Tuesday; 3 specifies Wednesday; 4 specifies Thursday; 5 specifies Friday; 6 specifies Saturday; and 7 specifies Sunday.</td>
-            <td>The weekday value indicating which day of the week is considered the ending day of the 'weekend', for calendar purposes.</td>
+            <td>[[Weekend]]</td>
+            <td>A List of 1 or more Integer value between 1 and 7: 1 specifies Monday; 2 specifies Tuesday; 3 specifies Wednesday; 4 specifies Thursday; 5 specifies Friday; 6 specifies Saturday; and 7 specifies Sunday.</td>
```
Comment (anba) [inline locale.html:139]:
````suggestion
            <td>A List of 1 or more integer values between 1 and 7: 1 specifies Monday; 2 specifies Tuesday; 3 specifies Wednesday; 4 specifies Thursday; 5 specifies Friday; 6 specifies Saturday; and 7 specifies Sunday.</td>
```

File: locale.html [hunk 5]
```diff
@@ -569,7 +564,7 @@ <h1>get Intl.Locale.prototype.textInfo</h1>
         1. Let _loc_ be the *this* value.
         1. Perform ? RequireInternalSlot(_loc_, [[InitializedLocale]]).
         1. Let _info_ be ! ObjectCreate(%Object.prototype%).
-        1. Let _dir_ be CharacterDirectionOfLocale(_loc_)
+        1. Let _dir_ be ! CharacterDirectionOfLocale(_loc_)
```
Comment (anba) [inline locale.html:567]:
```suggestion
        1. Let _dir_ be ! CharacterDirectionOfLocale(_loc_).
```

File: locale.html [hunk 6]
```diff
@@ -582,10 +577,10 @@ <h1>get Intl.Locale.prototype.weekInfo</h1>
         1. Let _loc_ be the *this* value.
         1. Perform ? RequireInternalSlot(_loc_, [[InitializedLocale]]).
         1. Let _info_ be ! ObjectCreate(%Object.prototype%).
-        1. Let _wi_ be WeekInfoOfLocale(_loc_)
+        1. Let _wi_ be ! WeekInfoOfLocale(_loc_)
```
Comment (anba) [inline locale.html:580]:
```suggestion
        1. Let _wi_ be ! WeekInfoOfLocale(_loc_).
```

**Review by anba — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -96,13 +96,17 @@ Add methods to Intl to get object to contains group of information:
...
-// {firstDay: 7, weekendStart: 6, weekendEnd: 7, minimalDays: 1}
+// {firstDay: 7, weekend: [6, 7], minimalDays: 1}
 enGB = new Intl.Locale("en-GB")
 enGB.weekInfo
-// {firstDay: 1, weekendStart: 6, weekendEnd: 7, minimalDays: 4}
+// {firstDay: 1, weekend: [6, 7], minimalDays: 4}
+
+let msBN = new Intl.Locale("ms-BN")
+msBN.weekInfo
+// {firstDay: 7, weekend: [5, 7], minimalDays: 1}  // Burnie weekend is Friday and Sunday but not Saturday
```
Comment (sffc) [inline README.md:109]:
```suggestion
// {firstDay: 7, weekend: [5, 7], minimalDays: 1}  // Brunei weekend is Friday and Sunday but not Saturday 
```

**Comment by sffc** [comment]
This PR was approved in 2021-09-09 TC39-TG2, modulo the editorial review comments: https://github.com/tc39/ecma402/blob/master/meetings/notes-2021-09-09.md#change-weekinfo-to-express-non-continouse-sic-weekend

**Pull request merged**